src/cpu/o3/iew.cc

```
@@ -111,11 +111,12 @@ IEW::IEW(CPU *_cpu, const BaseO3CPUParams &params)
     // Instruction queue needs the queue between issue and execute.
     instQueue.setIssueToExecuteQueue(&issueToExecQueue);
 
-    for (ThreadID tid = 0; tid < MaxThreads; tid++) {
-        dispatchStatus[tid] = Running;
+    for (ThreadID tid = 0; tid < numThreads; tid++) {
         fetchRedirect[tid] = false;
     }
 
+    assert(renameToIEWDelay == 1);
+
```

Cppcheck (2.19.0)

[error] Reference to temporary returned.

(returnTempReference)